public/app/core/components/Select/DashboardPicker.tsx

```diff
@@ -8,64 +9,40 @@ import { DashboardSearchHit, DashboardDTO } from 'app/types';
 
 export interface Props {
   onSelected: (dashboard: DashboardDTO) => void;
-  currentDashboardId?: SelectableValue<number>;
+  currentDashboard?: SelectableValue<number>;
   size?: FormInputSize;
+  isClearable?: boolean;
 }
 
-export interface State {
-  isLoading: boolean;
-}
-
-export class DashboardPicker extends PureComponent<Props, State> {
-  debouncedSearch: any;
-
-  static defaultProps = {
-    size: 'md',
-  };
-
-  constructor(props: Props) {
-    super(props);
-
-    this.state = {
-      isLoading: false,
-    };
-
-    this.debouncedSearch = debounce(this.getDashboards, 300, {
-      leading: true,
-      trailing: true,
-    });
-  }
-
-  getDashboards = (query = '') => {
-    this.setState({ isLoading: true });
+export const DashboardPicker: FC<Props> = ({ onSelected, currentDashboard, size = 'md', isClearable = false }) => {
+  const getDashboards = (query = '') => {
     return backendSrv.search({ type: 'dash-db', query }).then((result: DashboardSearchHit[]) => {
-      const dashboards = result.map((item: DashboardSearchHit) => ({
+      return result.map((item: DashboardSearchHit) => ({
         id: item.id,
         value: item.id,
         label: `${item.folderTitle ? item.folderTitle : 'General'}/${item.title}`,
       }));
-
-      this.setState({ isLoading: false });
-      return dashboards;
     });
   };
 
-  render() {
-    const { size, onSelected, currentDashboardId } = this.props;
-    const { isLoading } = this.state;
-
-    return (
-      <Forms.AsyncSelect
-        size={size}
-        isLoading={isLoading}
-        isClearable={true}
-        defaultOptions={true}
-        loadOptions={this.debouncedSearch}
-        onChange={onSelected}
-        placeholder="Select dashboard"
-        noOptionsMessage={'No dashboards found'}
-        value={currentDashboardId}
-      />
-    );
-  }
-}
+  const debouncedSearch = debounce(getDashboards, 300, {
+    leading: true,
+    trailing: true,
+  });
+
+  const [state, fetch] = useAsyncFn(debouncedSearch, []);
```

Comment:
`fetch` may be a misleading name here I think. Can you rename it to `searchDashboards` for instance?